stackslib/src/chainstate/nakamoto/tenure.rs

```diff
@@ -595,6 +595,53 @@ impl NakamotoChainState {
         Ok(tenure_opt)
     }
 
+    /// Get the tenure change that was active for a given block header
+    /// If a tenure change occurred during this block, it will be returned
+    pub fn get_tenure_for_block(
+        headers_conn: &Connection,
+        block_header: &StacksHeaderInfo,
+    ) -> Result<NakamotoTenure, ChainstateError> {
+        let sql = "SELECT * FROM nakamoto_tenures WHERE block_id = ? LIMIT 1";
+        let tenure_opt: Option<NakamotoTenure> =
+            query_row(headers_conn, sql, &[block_header.index_block_hash()])?;
+        if let Some(tenure) = tenure_opt {
+            return Ok(tenure);
+        }
+        // there wasn't a tenure change at that block, so we need to figure out the active tenure
+        //  use the "tenure height" to query for `num_blocks_confirmed`
+        let block_height = block_header.stacks_block_height;
+        let tenure_start_height = Self::get_nakamoto_tenure_start_block_header(
+            headers_conn,
+            &block_header.consensus_hash,
+        )?
+        .ok_or_else(|| ChainstateError::NoSuchBlockError)?
+        .stacks_block_height;
+        let blocks_confirmed = u64_to_sql(block_height.saturating_sub(tenure_start_height))?;
+        // querying by blocks confirmed doesn't work if cause is blockfound,
+        // so don't try and instead failback to directly querying it
+        let sql = "SELECT * FROM nakamoto_tenures WHERE tenure_id_consensus_hash = ?
+                                                    AND num_blocks_confirmed <= ?
+                                                    AND cause <> ?
```

Comment:
I think `!=` works in place of `<>`

Reply:
Yes, I default to `<>` because it's ISO standard SQL and `!=` isn't. But if you prefer `!=`, I can change it.